src/celeritas/neutron/model/detail/NuclearZoneBuilder.hh

```diff
@@ -0,0 +1,410 @@
+ * \f[
+     R = [ 1.16 * A^{1/3} - 1.3456 / A^{1/3} ] \cdot R_{scale}
+   \f]
+ * where \f$ R_{scale} \f$ is a configurable parameter in [femtometer], while
+ * \f$ R = 1.2 A^{1/3} \cdot R_{scale} \f$ (default) with a single parameter.
+ */
+real_type NuclearZoneBuilder::calc_nuclear_radius(AtomicMassNumber a)
+{
+    // Nuclear radius computed from A
+    real_type nuclear_radius{0};
+    auto amass = a.get();
+
+    if (amass > 4)
+    {
+        real_type cbrt_a = std::cbrt(static_cast<real_type>(amass));
+        real_type par_a = (options_.use_two_params ? 1.16 : 1.2);
+        real_type par_b = (options_.use_two_params ? -1.3456 : 0);
+        nuclear_radius = options_.radius_scale
+                         * (par_a * cbrt_a + par_b / cbrt_a);
+    }
+    else
+    {
+        nuclear_radius = options_.radius_small
+                         * (amass == 4 ? options_.radius_alpha : 1);
+    }
+
+    return nuclear_radius;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Integrate the Woods-Saxon potential, \f$ V(r) \f$,
+ *
+ * \f[
+     V(r) = frac{V_{o}}{1 + e^{\frac{r - R}{a}}}
+   \f]
+ * numerically over the volume from \f$ r_{min} \f$ to \f$ r_{rmax} \f$, where
+ * \f$ V_{o}, R, a\f$ are the potential well depth, nuclear radius, and
+ * surface thickness (skin depth), respectively.
+ */
+real_type
+NuclearZoneBuilder::integrate_woods_saxon(real_type rmin,
+                                          real_type rmax,
+                                          real_type nuclear_radius) const
+{
+    real_type skin_ratio = nuclear_radius / skin_depth_;
+
+    real_type ws_shift = 2 * skin_ratio;
+    real_type delta_r = rmax - rmin;
+
+    // Average value of \f$ r^{2} V(r) dr \f$ at boundaries
+    real_type integral = half() * delta_r
+                         * (rmin * (rmin + ws_shift) / (1 + std::exp(rmin))
+                            + rmax * (rmax + ws_shift) / (1 + std::exp(rmax)));
+
+    real_type result = integrate_potential(rmin, delta_r, integral, ws_shift);
+
+    return ipow<3>(skin_depth_)
+           * (result
+              + ipow<2>(skin_ratio)
+                    * std::log((1 + std::exp(-rmin)) / (1 + std::exp(-rmax))));
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Integrate the Gaussian potential function from \f$ r_1 \f$ to \f$ r_2 \f$.
+ */
+real_type NuclearZoneBuilder::integrate_gaussian(real_type rmin,
+                                                 real_type rmax,
+                                                 real_type gauss_radius) const
+{
+    real_type delta_r = rmax - rmin;
+    real_type rmin_sq = ipow<2>(rmin);
+    real_type rmax_sq = ipow<2>(rmax);
+
+    real_type integral
+        = this->half() * delta_r
+          * (rmin_sq * std::exp(-rmin_sq) + rmax_sq * std::exp(-rmax_sq));
+
+    real_type result = integrate_potential(false, rmin, delta_r, integral);
```

Comment:
I think this is wrong: `false` is being promoted to `rmin = 0`, `rmin` is being passed as `delta_r`, etc.

Reply:
Nice catch!  This is definitely a bug (left over from the earlier implementation).  I also realize that this Gaussian potential function is not tested (i.e, any A in [5, 11]), so the bug was not caught in validation (with respect to Geant4) in the test.  I will add a test case.